javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -490,15 +491,23 @@ default T get() {
      *
      * @return {@code Some(traversable)} or {@code None} if this is empty.
      */
-    Option<? extends Traversable<T>> initOption();
+    default Option<? extends Traversable<T>> initOption() {
+        return whenNotEmpty(this::init);
+    }
+
+    default <R> Option<R> whenNotEmpty(Supplier<R> supplier) {
```

Comment:
Yes, we should hide it. A good place is within javaslang.collection.Collections as static method.